### ISSUE TITLE
date picker cannot be focused with TAB key when zoom-in 400%

### ISSUE THREAD

**Issue by MingjieZhu** [vaadin/flow-components#4809]
### Description

add a date picker or a date-time picker. 
Zoom-in to 400% as 508 compliant requirements.
Press TAB key to go though each fields/buttons.
the focus skips date picker. (works when running 100%) 

### Expected outcome

expect TAB key will go through all fields including the date picker.

### Minimal reproducible example

DatePicker dField = new DatePicker();
			dField.setId("FC-" + fcField.getFcId() );
			dField.setLabel(fcField.getFldName());
			dField.setRequired(fcField.getRequired());
			dField.setValue(value == null ? null : DateUtils.getLocalDate(value));
			dField.addValueChangeListener(e -> {
				isChanged = true;			
			});
			dField.setWidth("100%");
			add(dField);

### Steps to reproduce

1. add a text field
2. add a date picker
3. run the app in browser (width must be 1280 pix as 508 compliant requirement)
4. zoom in 400%
5. TAB to the text field
6. cannot tab to the date picker

### Environment

Vaadin version(s): 23
OS: windows 11. Chrome 111

### Browsers

Chrome

**Comment by web-padawan** [comment]
Zooming the page decreases the viewport width, e.g. when changing browser width to 600px and then zoom to 150%, the actual width is 400px. This matches the [fullscreen media query](https://github.com/vaadin/web-components/blob/64f6da7a0aae7981671b2ea07d96ef80b5c35c2c/packages/date-picker/src/vaadin-date-picker-mixin.js#L125-L127) `(max-width: 420px), (max-height: 420px)`. 

Then, when using fullscreen mode, attempting to focus an input causes blur, which is mentioned in the [comment](https://github.com/vaadin/web-components/blob/64f6da7a0aae7981671b2ea07d96ef80b5c35c2c/packages/date-picker/src/vaadin-date-picker-mixin.js#L740-L741). As a workaround, it's possible to use `autoOpenDisabled` since https://github.com/vaadin/web-components/pull/3165. 

I will investigate if we could relax `_noInput` behavior to avoid blurring the `input` element on focus.

**Comment by rolfsmeds** [comment]
Could distinguishing between pointer-click and keyboard-focus help here? If the field is focused by keyboard, we know that it's a small-viewport keyboard-driven user, which means that the overlay shouldn't open automatically, but on space / arrow-down, after which focus needs to move into the overlay.

**Comment by web-padawan** [comment]
I think we could generally 

1. Allow focusing the `<input>` in `fullscreen` mode even if auto open is enabled, as keyboard focus does not open the overlay automatically. This would fix the original issue,
2. When opening the overlay in `noInput` / `fullscreen` mode, always focus date element and blur the input,
3. When closing the overlay in `fullscreen` mode, allow to restore focus (do not blur the input).

So basically, the current `noInput` mode would only be used on iOS / Android, but not on desktop browsers.

**Comment by rolfsmeds** [comment]
Sounds good. I suppose this would also solve it for small-screen iOS/Android devices with keyboards (e.g. bluetooth)?

**Comment by web-padawan** [comment]
I'm not sure if we can reliably handle the keyboards on iOS / Android devices as I'm not aware of a reliable way to detect whether the `keydown` event is from virtual or external keyboard. While we have explicit logic to disable virtual keyboard by setting `inputMode` since https://github.com/vaadin/web-components/pull/3253, I'm not sure if that's sufficient. 

So this would likely need more testing with a real bluetooth keyboard, and I don't have one at my disposal.

**Comment by rolfsmeds** [comment]
I was thinking that it wouldn't matter – that it only matters how the field is _focused_. If you're interacting with the UI primarily via an actual keyboard, you probably _tab_ into it, whereas if you're interacting primarily via touch, you'd probably _tap_ it.
